azurerm/internal/services/managedapplication/validate/managed_application_definition_test.go

```diff
@@ -0,0 +1,209 @@
+package validate
+
+import (
+	"testing"
+)
+
+func TestManagedApplicationDefinitionName(t *testing.T) {
+	testData := []struct {
+		input    string
+		expected bool
+	}{
+		{
+			// empty
+			input:    "",
+			expected: false,
+		},
+		{
+			// basic example
+			input:    "hello",
+			expected: true,
+		},
+		{
+			// can't start with an underscore
+			input:    "_hello",
+			expected: false,
+		},
+		{
+			// can't end with a dash
+			input:    "hello-",
+			expected: false,
+		},
+		{
+			// can't contain an exclamation mark
+			input:    "hello!",
+			expected: false,
+		},
+		{
+			// dash in the middle
+			input:    "malcolm-in-the-middle",
+			expected: false,
+		},
+		{
+			// can't end with a period
+			input:    "hello.",
+			expected: false,
+		},
+		{
+			// 63 chars
+			input:    "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijk",
+			expected: true,
+		},
+		{
+			// 64 chars
+			input:    "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijkj",
+			expected: true,
+		},
+		{
+			// 65 chars
+			input:    "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijkja",
+			expected: false,
+		},
+	}
+
+	for _, v := range testData {
+		t.Logf("[DEBUG] Testing %q..", v.input)
+
+		_, errors := ManagedApplicationDefinitionName(v.input, "name")
+		actual := len(errors) == 0
+		if v.expected != actual {
+			t.Fatalf("Expected %t but got %t", v.expected, actual)
+		}
+	}
+}
+
+func TestManagedApplicationDefinitionDisplayName(t *testing.T) {
+	testData := []struct {
+		input    string
+		expected bool
+	}{
+		{
+			// empty
+			input:    "",
+			expected: false,
+		},
+		{
+			// basic example
```

Comment:
Don't copy paste the test case from other test, each test case need its own purpose.

Reply:
updated